CS3219-AY2223S1/cs3219-project-ay2223s1-g12
# Create socket.io client instance when find match button is clicked

Fixes #4 , #37

Summary: 
- Currently, there is no connection between the frontend views and backend 
- This PR creates a socket.io client to connect to the socket.io server whenever a match button is clicked 
- This PR also routes the user to the countdown timer the match button is clicked 

Whenever a match button is clicked, a socket.io client is instantiated and emits the event 
<img width="399" alt="image" src="https://user-images.githubusercontent.com/77088875/189468667-6771d541-0864-4640-a002-1805b05d323f.png">

The server receives it and logs the event onto the console 
<img width="857" alt="image" src="https://user-images.githubusercontent.com/77088875/189468622-34127a97-0fc5-4a72-876b-afd23aaf2199.png">


How to test:
- cd to `matching-service` and run `node index.js`
- open a new terminal
- cd to `frontend` and run `npm start` (run `npm install` beforehand to install dependencies)
- navigate to `http://localhost:3000/selectquestiondifficulty`
- click on a button
- check that you are routed to the countdown timer 
- go back to the matching service page and look at the console logs to check if event has been received by server
